src/Symfony/Component/PropertyAccess/PropertyPath.php

```diff
@@ -202,4 +219,16 @@ public function isIndex(int $index)
 
         return $this->isIndex[$index];
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isOptional(int $index)
+    {
+        if (!isset($this->isOptional[$index])) {
+            throw new OutOfBoundsException(sprintf('The index %s is not within the property path', $index));
```

Comment:
Missing dot at the end of the message.

Reply:
This is the same message as the other `isIndex`, `isProperty` methods and those don't have a dot as well. Maybe handle it in an other PR? to fix those at the same time.

Reply:
This was fixed in sf 5.0 fixing it.